src/Sparrow/Memory.cs

```diff
@@ -261,5 +263,22 @@ public static T Read<T>(byte* ptr)
         {
             return Unsafe.Read<T>(ptr);
         }
+
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        public static void Discard(void* baseAddress, long size)
+        {
+            if (PlatformDetails.CanDiscardMemory)
+            {
+                // We explicitely ignore the return codes because even if it fails, from all uses and purposes we dont care. 
+                if (PlatformDetails.RunningOnPosix)
+                {
+                    Syscall.madvise(new IntPtr(baseAddress), new UIntPtr((ulong)size), MAdvFlags.MADV_DONTNEED);
+                }
+                else
+                {
+                    Win32MemoryProtectMethods.DiscardVirtualMemory(baseAddress, new UIntPtr((ulong)size));
```

Comment:
Can we log this if we failed?
(Debug level?)

Reply:
We could. Just in debug mode though, not that it makes any difference to know though. It can fail and we will still working properly.

Reply:
Maybe we could try it once on server startup and if we fail then log and create an alert?

Not sure if this call might be expensive but maybe if we fail a few times consecutively then let's skip this call completely.

Reply:
WRT my 2nd comment - already answered in https://github.com/ravendb/ravendb/pull/8160#discussion_r239216380